drizzlepac/hapsequencer.py

```diff
@@ -1411,3 +1419,62 @@ def _verify_sci_hdrname(filename):
     if closefits:
         fhdu.close()
         del fhdu
+
+# ------------------------------------------------------------------------------
+
+
+def reset_keywords_apriori(filename):
+    """
+    Check the SCI extension(s) of the output FLT/FLC and DRZ/DRC files.  If the active
+    WCS solution is 'a priori', there should not be any valid values for NMATCHES,
+    RMS_RA/RMS_DEC, FITGEOM, and CRDER1/CRDER2 as these keywords are associated with 
+    'a posteriori' solutions.  Ensure the WCSTYPE is based upon the active WCSNAME.
+
+    Parameters
+    ----------
+    filename : str
+        Name of file to evaluate
+
+    Returns
+    -------
+    None
+    """
+    wcs_keys = {'NMATCHES': 0, 'RMS_RA': 0.0, 'RMS_DEC': 0.0, 'FITGEOM': '', 
+                'CRDER1': 0.0, 'CRDER2': 0.0}
+
+    # Determine the number of SCI extensions
+    num_sci, extname = util.count_sci_extensions(filename)
+    extname_list = [(extname, x + 1) for x in range(num_sci)]
+
+    # Open the file in anticipation of updating 'a priori' solution associated keywords
+    hdu = fits.open(filename, mode='update')
+
+    # Loop over the SCI extensions in the file
+    for ext in extname_list:
+        active_wcsname = hdu[ext].header['WCSNAME']
+
+        # An 'a priori' solution is of the form <Starting WCS>-<Astrometric Catalog>
+        # Catalogs: GSC240, HSC30, GAIAD1, GAIADR2, and GAIAeDR3
+        wcsname_list = active_wcsname.split('-')
+
+        # If the WCS solution is 'a priori' ensure any keywords in wcs_keys are set to default values
+        is_apriori = len(wcsname_list) == 2 and wcsname_list[1].upper().startswith(('GSC', 'HSC', 'GAIA'))
+        if is_apriori:
+            try:
+                # Make sure the WCSTYPE is consistent with the active 'a priori' WCSNAME
+                if hdu[ext].header['WCSTYPE']:
+                    hdu[ext].header['WCSTYPE'] = updatehdr.interpret_wcsname_type(active_wcsname)
+            except KeyError:
+                pass
+
+            # Loop over the keyword values for this particular solution to update
+            for wkey, wvalue in wcs_keys.items():
+                try:
+                    if hdu[ext].header[wkey]:
+                        hdu[ext].header[wkey] = wvalue
+                except KeyError:
```

Comment:
I do not understand the purpose of this. It seems to me that this code does not do what the comment in the function description says:

```
If the active WCS solution is 'a priori', there should not be any valid values for NMATCHES,
RMS_RA/RMS_DEC, FITGEOM, and CRDER1/CRDER2 as these keywords are associated with 
'a posteriori' solutions.
```
So, instead of setting these keyword values to some obviously wrong values, I think tyou should delete them as the statement above says:
```suggestion
            # Loop over the keyword values for this particular solution to update
            for wkey, wvalue in wcs_keys.items():
                if wkey in hdu[ext].header:
                    del hdu[ext].header[wkey]
```

Reply:
Deleting the keywords was offered as a solution, but INS requested they be set to default values instead.

Reply:
Sigh... But then the comment below the function declaration is wrong as it does not describe what this function is actually doing.

Reply:
And these keywords do not make any sense for the drizzled images. These values are different or each input image.  How does one combine this into a single value to be written to a drizzled image, a priori or not? It just feels wrong.

Reply:
if INS wants the keywords at all times and regardless of the content, then you do not need `try` at all. This is much simpler:

```python
hdu[ext].header.update(wcs_keys)
```

Reply:
Deleting the keywords is an acceptable solution.  The goal was simply to not confuse users with fit rms values, etc when there was in fact no catalog fit.